src/app/core/learning-object-module/revisions/revisions.service.ts

```diff
@@ -1,9 +1,63 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { REVISION_ROUTES } from '../revisions/revisions.routes';
 
 @Injectable({
   providedIn: 'root'
 })
 export class RevisionsService {
 
-  constructor() { }
+  constructor(private http: HttpClient) {}
+  
+  /**
+   * Creates a Revision of an existing learning object
+   *
+   * @param cuid the CUID of the learning object to create a revision of
+   */
+  async createRevision(cuid: string): Promise<any> {
+    const route = REVISION_ROUTES.CREATE_REVISION(cuid);
+    const response = await this.http
+      .post(route, {}, { withCredentials: true })
+      .pipe(catchError(this.handleError))
+      .toPromise();
+    return response;
+  }
```

Comment:
The calls to all of these functions will also need to be updated in order to properly build the client.